aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -1550,17 +1551,18 @@ inline void baddbmm_cpu_kernel(const Tensor& result, const Tensor& self, const T
           auto r2 = r1[i];
           auto s2 = s1[i];
           for (const auto j : c10::irange(js)) {
-            scalar_t &r = r2[j];
+            opmath_t acc_value = 0;//is_bmm ? opmath_t(0) : opmath_t(r2[j]);
+            for (const auto k : c10::irange(ks)) {
+              acc_value += s2[k] * m1[k][j];
+            }
             if (is_bmm) {
-              r = 0;
-              for (const auto k : c10::irange(ks)) {
-                r += s2[k] * m1[k][j];
-              }
+              r2[j] = acc_value;
             } else {
               // For beta == 0, the r's value will be ignored, especially for nan value.
-              r = beta == scalar_t(0) ? scalar_t(0) : beta * r;
-              for (const auto k : c10::irange(ks)) {
-                r += alpha * s2[k] * m1[k][j];
+              if (beta == scalar_t{0}) {
+                r2[j] = alpha * acc_value;
+              } else {
+                r2[j] = r2[j] * beta + alpha * acc_value;
```

Comment:
The assignments of `alpha` and `beta` above should be changed to `.to<opmath_t>()`

Reply:
Done.